### PR TITLE
Dart compile fix

### PR DESCRIPTION
**Pull request by tbosch**

### CODE REVIEW

File: modules/angular2/src/dom/browser_adapter.dart [hunk 1]
```diff
@@ -98,17 +98,19 @@ final _keyCodeToKeyMap = const {
 class BrowserDomAdapter extends GenericBrowserDomAdapter {
   js.JsFunction _setProperty;
   js.JsFunction _getProperty;
-  js.JsFunction _hasProperty;
   BrowserDomAdapter() {
-    _setProperty = js.context.callMethod('eval', ['(function(el, prop, value) { el[prop] = value; })']);
+    _setProperty = js.context.callMethod('eval', ['(function(el, prop, value) { if (prop in el) el[prop] = value; })']);
```
Comment (yjbanov) [inline modules/angular2/src/dom/browser_adapter.dart]: Do we intend this to be a permanent solution? This code is not CSP-compliant.

File: modules/angular2/src/dom/html_adapter.dart [hunk 2]
```diff
@@ -11,7 +11,9 @@ class Html5LibDomAdapter implements DomAdapter {
   }
 
   hasProperty(element, String name) {
-    // This is needed for serverside compile to generate the right getters/setters...
+    // This is needed for serverside compile to generate the right getters/setters.
+    // TODO: change this once we have property schema support.
+    // Attention: Keep this in sync with browser_adapter.dart!
```
Comment (yjbanov) [inline modules/angular2/src/dom/html_adapter.dart]: Why does it need to be in sync? `BrowserDomAdapter.hasProperty` wouldn't be called when transformers are used. I think there are parts of `BrowserDomAdapter` that are always necessary and parts that are only needed w/o transformers. I think we should split it up so we can make sure the non-necessary part can be turned off when transformers are used.

File: modules/angular2/src/dom/browser_adapter.dart (hunk 1 above)
Comment (tbosch) [inline modules/angular2/src/dom/browser_adapter.dart]: True. For CSP mode we could have a slow path the uses `JsObject`. Created https://github.com/angular/angular/issues/3025 for this.

File: modules/angular2/src/dom/html_adapter.dart (hunk 2 above)
Comment (tbosch) [inline modules/angular2/src/dom/html_adapter.dart]: Based on this flag, we are not only generating change detectors but also the `ChangeRecords`, which are not generated by the transformer. I.e. `ChangeRecords` are recreated when the application bootstraps via a compiler run.

**Comment by tbosch** [comment]
Closed via b3a763a718610aa322848aef009b70c112f33377

**Pull request closed without merging**